polkadot/runtime/parachains/src/scheduler.rs

```diff
@@ -345,11 +354,16 @@ impl<T: Config> Pallet<T> {
 	}
 
 	// on new session
-	fn maybe_resize_claim_queue(old_core_count: u32, new_core_count: u32) {
-		if new_core_count < old_core_count {
+	fn maybe_resize_claim_queue() {
+		let cq = ClaimQueue::<T>::get();
+		let Some((old_max_core, _)) = cq.last_key_value() else {return};
+		let config = configuration::ActiveConfig::<T>::get();
+		let new_core_count = Self::expected_claim_queue_len(&config.scheduler_params);
+
+		if new_core_count < (old_max_core.0 + 1) {
```

Comment:
```suggestion
		if new_core_count < cq.len() {
```

Reply:
It can't be simply `.len()` because the claim queue could have holes (for cores that don't have any assignments).